cmd/jaeger/internal/integration/span_reader.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/ports"
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+var (
+	_ spanstore.Reader = (*spanReader)(nil)
+	_ io.Closer        = (*spanReader)(nil)
+)
+
+// SpanReader retrieve span data from Jaeger-v2 query with api_v2.QueryServiceClient.
+type spanReader struct {
+	Port int
+
+	clientConn *grpc.ClientConn
+	client     api_v2.QueryServiceClient
+}
+
+func createSpanReader(port int) *spanReader {
+	return &spanReader{
+		Port: port,
+	}
```

Comment:
you may be missing coverage instruction and upload

also, wonder if https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/grpc/shared/grpc_client_test.go could be repurposed for testing these span reader/writer, since integration testing of them will not provide full coverage

Reply:
https://github.com/jaegertracing/jaeger/pull/5322#issuecomment-2033748607 
Have you read this comment? I think the main reason is that the gRPC (v2) integration test is failing due to trace data translation between Jaeger proto and OTLP tag attributes. Jaeger proto's BINARY tag type is translated into STRING type in the OTLP format and remains as STRING type when translated back into Jaeger proto.

Reply:
I think it's a bug in the translator. OTLP supports bytes in the AnyValue, the translator did not have to convert it to a string.
```
message AnyValue {
  // The value is one of the listed fields. It is valid for all values to be unspecified
  // in which case this AnyValue is considered to be "empty".
  oneof value {
    string string_value = 1;
    bool bool_value = 2;
    int64 int_value = 3;
    double double_value = 4;
    ArrayValue array_value = 5;
    KeyValueList kvlist_value = 6;
    bytes bytes_value = 7;
  }
}
```

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32204